### PR TITLE
Feature: Display asset types from the backend

### PR DESCRIPTION
**Pull request by andela-fmustapha**
## What does this PR do?
This PR adds an Asset Types side menu option and it also ensures users can view all asset types
from the background.
 
## Description of Task to be completed?
- Use the ART Backend API endpoint to show all asset types
- AS a USER
   WHEN I click on the asset types link on the sidemenu
    - It should fetch all assets from the backend and display paginated results in a table
    - Add an action colum which will contain a Show, Edit Icon/button for each assets
    - Add a button for adding new assets on the assets page

## How should this be manually tested?
- Pull this branch and click on the `Asset Types` menu option to view asset types

## What are the relevant pivotal tracker stories?
[158491088](https://www.pivotaltracker.com/story/show/158491088)

## Any background context you want to add?
-There was no Asset Types menu option initially.

## Important notes
n/a
## Packages installed
n/a
## Deployment note
n/a
## Related PRs branch | PR (branch_name) | (pr_link)
n/a
## Screenshots
screen shot 2018-06-29 at 12 17 48 pm

### CODE REVIEW

File: src/components/TableRowComponent.jsx [hunk 1]
```diff
@@ -1,22 +1,51 @@
...
           {props.data[heading]}
         </Table.Cell>
+        )
+      )
       }
-      )}
+    {
+        props.action ?
+          <Table.Cell>
+            {action}
```
[the remark this answers is not in the document]
Comment (andela-fmustapha) [reply src/components/TableRowComponent.jsx]: Please could you clarify your question, I'm not sure I understand.

[the remark this answers is not in the document]
Comment (sachungo) [reply src/components/TableRowComponent.jsx]: the ternary check is using `props.action` yet the value is `action`. I was asking if the value should also be `props.action`

File: src/components/AssetTypesComponent.jsx [hunk 2]
```diff
@@ -1,98 +1,161 @@
...
-              <Table.Row>
-                <Table.HeaderCell colSpan='3'>
-                  {
-                    (this.emptyAssetTypeCheck()) ? '' :
+        {(this.emptyAssetTypeCheck()) ?
+          <LoaderComponent
+            size="large"
+            loadingText="loading..."
+          /> :
+          <Container>
```
Comment (sachungo) [inline src/components/AssetTypesComponent.jsx]: The ternary is extra long making it hard to read the code, especially the `<Container>...</Container>` part.  Therefore, I'm going to ask you again to make some changes to the render function. I.e.
```
render() {
  if (this.props.isLoading) {
    return (...);
  }

  if (!this.props.isLoading && _.isEmpty(this.props.assetTypes)) {
    return (...);
  }

  if (this.emptyAssetTypeCheck()) {
    return (
      <SideMenuComponent>
        <LoaderComponent
          size="large"
          loadingText="loading..."
        />
      </SideMenuComponent>
    );
  }

  return (
    <SideMenuComponent>
      <Container>
        ...
        ...
      </Container>
    </SideMenuComponent>
  );
}
```

**Note** that use of `else if` is unnecessary due to the return statements

**Comment by sachungo** [comment]
@andela-fmustapha please fix the conflicts and action the feedback 🙂

File: src/_actions/assetType.action.js [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
-    });
-  }
+/**
+ * load Asset Types thunk
+ *
+ * @return dispatch type and payload
+ */
+export const loadAssetTypeAction = () => (dispatch) => {
+  dispatch({ type: LOADING_ASSET_TYPE });
+  return axios.get('https://my-json-server.typicode.com/HawiCaesar/jsonplaceholders-demo/types')
```
Comment (sachungo) [inline src/_actions/assetType.action.js]: Note: Mock URL to be removed in a later PR

**Pull request merged**